### PR TITLE
http: Update godocs

### PR DESCRIPTION
**Pull request by abhinav**
I took a pass over the transport/http godocs to make them a bit more readable.
Added examples in some places and documented the wire format.

@yarpc/golang

### CODE REVIEW

File: transport/http/doc.go [hunk 1]
```diff
@@ -18,9 +18,45 @@
...
+// requests. Connections will remain open until all clients have disconnected.
+//
+// Wire Representation
+//
+// YARPC requests and responses are sent as plain HTTP requests and responses.
+// YARPC metadata is sent inside reserved HTTP headers. Application headers
+// for requests and responses are sent as HTTP headers with the header names
+// prefixed with a pre-defined string. See Constants for more information on
+// the names of these headers. The request and response bodies are sent as-is
+// in the HTTP request or response body.
```
Comment (kriskowal) [inline transport/http/doc.go]: maybe link yarpc/yarpc/properties.md

Comment (kriskowal) [inline transport/http/doc.go]: https://github.com/yarpc/yarpc/blob/master/properties.md

File: transport/http/outbound.go [hunk 2]
```diff
@@ -94,24 +104,31 @@ func NewOutbound(chooser peer.Chooser, opts ...OutboundOption) *Outbound {
...
-	o := t.NewOutbound(peerchooser.NewSingle(hostport.PeerIdentifier(parsedURL.Host), t))
-	o.setURLTemplate(URL)
+
+	chooser := peerchooser.NewSingle(hostport.PeerIdentifier(parsedURL.Host), t)
+	o := t.NewOutbound(chooser)
 	for _, opt := range opts {
 		opt(o)
 	}
+
+	o.setURLTemplate(uri)
```
Comment (kriskowal) [inline transport/http/outbound.go]: :ok:

File: transport/http/doc.go [hunk 3]
```diff
@@ -18,9 +18,45 @@
...
 
-// Package http implements the HTTP inbound and outbound transports for YARPC.
+// Package http implements a YARPC transport based on the HTTP/1.1 protocol.
+// The HTTP transport provides first class support for Unary RPCs and
+// experimental support for Oneway RPCs.
 //
-// Note that the Close method for the HTTP Inbound does NOT immediately close
-// the ongoing connections. The connection will remain open until all clients
-// have disconnected.
+// Usage
```
Comment (breerly) [inline transport/http/doc.go]: This is great.

File: transport/http/doc.go (hunk 1 above)
Comment (abhinav) [inline transport/http/doc.go]: Added

image

**Pull request merged**